Part6

austin-lolli opened the pull request:
Project 4 Part 6 submission

austin-lolli committed:
First Commit, declared apply functions in UDT's

austin-lolli committed:
Step 3 Commit

austin-lolli committed:
Step 4 commit

austin-lolli committed:
Step 4 updated commit

austin-lolli committed:
Step 5 Commit

austin-lolli committed:
step 6 mid std::function issue fix

austin-lolli committed:
Project 4 Part 6 Submission

austin-lolli committed:
Fixed spacing in test value prints

matkatmusic commented on main.cpp:
```diff
@@ -131,6 +127,27 @@ struct FloatType
+            {
+                return callable(*heapFloat); 
+            }
+            else
+            {
+                return *this;
+            }
```
You have no code that runs after the `else` block.  Therefore you do not need this `else` keyword.  You did it correctly in the `apply()` below.
- [ ] revise

matkatmusic commented on main.cpp:
```diff
@@ -178,6 +195,27 @@ struct DoubleType
+    
+        DoubleType& apply(std::function<DoubleType&(double&)> callable)
+        {
+            if(callable)
+            {
+                return callable(*heapDub); 
+            }
+            else
+            {
+                return *this;
+            }
```
You have no code that runs after the `else` block.  Therefore you do not need this `else` keyword
- [ ] revise

matkatmusic commented on main.cpp:
```diff
@@ -232,6 +271,29 @@ struct IntType
+
+        IntType& apply(std::function<IntType&(int&)> callable)
+        {
+            if(callable)
+            {
+                return callable(*heapInt); 
+            }
+            else
+            {
+                return *this;
+            }
```
You have no code that runs after the `else` block.  Therefore you do not need this `else` keyword
- [ ] revise

matkatmusic reviewed (changes requested):
minor change requested

Also, I said don't edit your previous main, yet you commented a bunch of it out.  uncomment it.

matkatmusic committed:
fixme: removed unnecessary else blocks from std::function apply
fixme: uncommented tests for parts 4 and 5 in int main (main now matches part 5 + lamdas)
